pkg/storage/util.go

```diff
@@ -106,23 +104,19 @@ func NamespaceKeyFunc(prefix string, obj runtime.Object) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	name := meta.GetName()
-	if msgs := validation.IsValidPathSegmentName(name); len(msgs) != 0 {
-		return "", fmt.Errorf("invalid name: %v", msgs)
-	}
-	return prefix + "/" + meta.GetNamespace() + "/" + name, nil
+	// We assume that NamespaceKeyFunc is used for read requests (List and Watch)
+	// and thus the object is correct and we don't perform any validation.
+	return prefix + "/" + meta.GetNamespace() + "/" + meta.GetName(), nil
 }
 
 func NoNamespaceKeyFunc(prefix string, obj runtime.Object) (string, error) {
 	meta, err := meta.Accessor(obj)
 	if err != nil {
 		return "", err
 	}
-	name := meta.GetName()
-	if msgs := validation.IsValidPathSegmentName(name); len(msgs) != 0 {
```

Comment:
And I'm in the wrong package.  I didn't realize we had clones of these methods in a second spot.  I haven't dug through what a bug here would mean.